### ISSUE TITLE
Hyperbolic Models Diverge for External Dataset

### ISSUE THREAD

**Issue by davidkartchner** [HazyResearch/KGEmb#1]
I'm trying to benchmark each of the hyperbolic embedding models on an external knowledge graph.  Performance seems to improve marginally for each of the first two epochs but then diverges (loss becomes nan, all metrics report "perfect" scores on test/eval).  I have trained an instance of each of the possible models in the repository and only RotH, RefH, and AttH diverge.  All of my data is formatted exactly as in the provided example datasets (including to_skip dictionary).

Your help in resolving this is greatly appreciated!  I am attaching my `train.log` file for [AttH](https://github.com/HazyResearch/KGEmb/files/4713670/train.log), [RefH](https://github.com/HazyResearch/KGEmb/files/4713677/train.log), and [RotH](https://github.com/HazyResearch/KGEmb/files/4713674/train.log) for your convenience.  I'm also happy to include a copy of the data directory I'm using if it would be helpful.

**Comment by ines-chami** [comment]
Hi, 
By looking at the training logs it seems that the training/validation losses are diverging and I suspect that gradients are too large, thus causing nans. I would suggest trying decreasing the learning rate and see if that solves the issue?
